pkg/minikube/notify/notify_test.go

```diff
@@ -146,39 +152,80 @@ func TestGetLatestVersionFromURLMalformed(t *testing.T) {
-	if len(outputBuffer.String()) != 0 {
-		t.Fatalf("Expected MaybePrintUpdateText to not output text as the current version is %s and version %s was served from URL but output was [%s]",
-			version.GetVersion(), latestVersionFromURL, outputBuffer.String())
+	var tc = []struct {
+		len                     int
+		wantUpdateNotification  bool
+		latestVersionFromURL    string
+		description             string
+		status                  bool
+		url                     string
+		lastUpdateCheckFilePath string
+	}{
+		{
+			len:                    1,
+			latestVersionFromURL:   "0.0.0-dev",
+			wantUpdateNotification: true,
+			description:            "latest version lower or equal",
+		},
+		{
+			len:                    0,
+			latestVersionFromURL:   "100.0.0-dev",
+			wantUpdateNotification: true,
+			description:            "latest version greater",
+			status:                 true,
+		},
+		{
+			len:                    1,
+			latestVersionFromURL:   "100.0.0-dev",
+			wantUpdateNotification: false,
+			description:            "notification unwanted",
+		},
+		{
+			len:                    1,
+			latestVersionFromURL:   "100.0.0-dev",
+			wantUpdateNotification: true,
+			description:            "bad url",
+			url:                    "this is not valid url",
+			status:                 true,
+		},
+		{
+			len:                     1,
+			latestVersionFromURL:    "10.0.0-dev",
+			wantUpdateNotification:  true,
+			description:             "bad lastUpdateCheckFilePath",
+			lastUpdateCheckFilePath: "/etc/passwd",
+			status:                  true,
+		},
 	}
 
-	// test that update text is printed if the latest version is greater than the current version
-	latestVersionFromURL = "100.0.0-dev"
-	handler = &URLHandlerCorrect{
-		releases: []Release{{Name: version.VersionPrefix + latestVersionFromURL}},
-	}
-	server = httptest.NewServer(handler)
-	defer server.Close()
-
-	MaybePrintUpdateText(server.URL, lastUpdateCheckFilePath)
-	if len(outputBuffer.String()) == 0 {
-		t.Fatalf("Expected MaybePrintUpdateText to output text as the current version is %s and version %s was served from URL but output was [%s]",
-			version.GetVersion(), latestVersionFromURL, outputBuffer.String())
+	viper.Set(config.ReminderWaitPeriodInHours, 24)
+	for _, test := range tc {
+		t.Run(test.description, func(t *testing.T) {
+			viper.Set(config.WantUpdateNotification, test.wantUpdateNotification)
+			lastUpdateCheckFilePath = filepath.Join(tempDir, "last_update_check")
+			if test.lastUpdateCheckFilePath != "" {
+				lastUpdateCheckFilePath = test.lastUpdateCheckFilePath
+			}
+			latestVersionFromURL := test.latestVersionFromURL
+			handler := &URLHandlerCorrect{
+				releases: []Release{{Name: version.VersionPrefix + latestVersionFromURL}},
+			}
+			server := httptest.NewServer(handler)
+			defer server.Close()
+			if test.url == "" {
+				test.url = server.URL
+			}
+			status := MaybePrintUpdateText(test.url, "/dev/null")
```

Comment:
Probably, better to create a tempdir here, and use it's path, because `/dev/null` will be a problem on windows.

Reply:
fixed. Thanks for the review.